trees/PrescriptiveTree.py

```diff
@@ -0,0 +1,369 @@
+        try:
+            t = t.astype(int)
+        except Exception as e:
+            print('The set of treatments must be discrete.')
+
+        assert (
+                min(t) == 0 and max(t) == len(set(t)) - 1
+        ), 'The set of treatments must be discrete starting from {0, 1, ...}'
+
+        # we also need to check on y and ipw/y_hat depending on the method chosen
+        y = self.check_y(X, y)
+        ipw, y_hat = self.check_helpers(X, ipw, y_hat)
+
+        # Raises ValueError if there is a column that has values other than 0 or 1
+        check_binary(X)
+
+        # Store the classes seen during fit
+        self.treatments_ = np.unique(t)
+
+        # keep original data
+        self.X_ = X
+        self.y_ = y
+        self.t_ = t
+
+        # Instantiate tree object here
+        self.tree = Tree(self.depth)
+
+        # Code for setting up and running the MIP goes here.
+        # Note that we are taking X and y as array-like objects
+        self.start_time = time.time()
+
+        if self.method == 'IPW':
+            self.primal = FlowOPT_IPW(
+                X,
+                t,
+                y,
+                ipw,
+                self.treatments,
+                tree,
+                self.X_col_labels,
+                self.time_limit,
+                self.num_threads
+            )
+
+        elif self.method == 'DM':
+            self.primal = FlowOPT_Robust(
+                X,
+                t,
+                y,
+                ipw,
+                y_hat,
+                False,
+                self.treatments,
+                tree,
+                X_col_labels,
+                self.time_limit,
+                self.num_threads
+            )
+
+        elif self.method == 'DR':
+            self.primal = FlowOPT_Robust(
+                X,
+                t,
+                y,
+                ipw,
+                y_hat,
+                True,
+                self.treatments,
+                tree,
+                X_col_labels,
+                self.time_limit,
+                self.num_threads
+            )
+
+        self.primal.create_main_problem()
+        self.primal.model.update()
+        self.primal.model.optimize()
+
+
+        self.end_time = time.time()
+        # solving_time or other potential parameters of interest can be stored
+        # within the class: self.solving_time
+        self.solving_time = self.end_time - self.start_time
+
+        # Here we will want to store these values and any other variables
+        # needed for making predictions later
+        self.b_value = self.primal.model.getAttr("X", self.primal.b)
+        self.w_value = self.primal.model.getAttr("X", self.primal.w)
+        self.p_value = self.primal.model.getAttr("X", self.primal.p)
+
+        # Return the classifier
+        return self
+
+    def predict(self, X):
+        """A reference implementation of a prediction for a classifier.
```

Comment:
Need a short description here along the lines of "Method for making predictions using a PrescriptiveTree classifier"